### ISSUE TITLE
cush.config.js

### ISSUE THREAD

**Issue by aleclarson** [aleclarson/cush#7]
```js
// for javascript bundles only
exports.js = function() {
  this.plugin('markdown');    // enable .md modules
  this.plugin([ /* ... */ ]); // enable multiple plugins at once
  this.set('jsxPragma', 'h'); // use preact
  this.nebu.use([
    // nebu plugins goes here
  ]);
  console.log('dev:', this.dev);   // know if `this` is a development bundle
  console.log('root:', __dirname); // the project root
};
```

Since the exported function is called once *per bundle*, you may want to declare some variables outside if recreating them is costly, but I think that will be rare.

Bundle plugins must exist in the `devDependencies` of the project's `package.json` if you want to activate them using `this.plugin`. You can use the `file:` protocol to reference local plugins, if necessary.

It might be cool to share this API with bundle plugins, so once you're familiar with bundle configuration, you're capable of making a bundle plugin, too! That would mean making this API a little more flexible by adding `this.hook` for package-loading and module-loading functions. Maybe bundle plugins could provide their own hooks, too!

I'm open to suggestions on other capabilities. 😄

**Comment by aleclarson** [comment]
The hook API:
```js
// Know when `cush.config.js` has been applied.
this.hook('config', function() {
  // `this` remains the same in here.
});

// Provide a hook.
let hook = this.hook('my-hook');
hook.run(a, b, c);
```
The provider API may be expanded in the future.

**Comment by aleclarson** [comment]
The `BundleConfig` class will always have these methods:
- `get(path: string)` access a bundle option
- `set(path: string, value: any)` update a bundle option
- `plugin(name: string | string[])` activate a plugin or plugins
- `hook(id: string, hook: ?function)` create a provider or hook into one
- `hookModules(ext: string, hook: function)` hook into the module loader
- `hookPackages(hook: function)` hook into the package loader

...and these properties:
- `dev: boolean`
- `target: string`

**Comment by aleclarson** [comment]
Implemented in d4025db59fc5e9267aadf31fdace98ab46d2e9c5 🎉
